readme.txt
==Title==
Bitcoin Payment Module for Gambio GX2

==Copyright==
(c) 2014, Commerce Coding
http://www.commerce-coding.de

==Current Version==
0.1.0

changelog.txt
==0.1.0 (09.02.2014)==
* initial release

src/new/callback/bitcoin/callback.php
<?php
/**
 * CommerceCoding Bitcoin Payment for Gambio GX2
 *
 * NOTICE OF LICENSE
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; version 2 of the License
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software
 * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
 *
 * @copyright   Copyright (c) 2013 CommerceCoding (http://www.commerce-coding.de)
 * @author      Alexander Diebler
 * @license     http://opensource.org/licenses/GPL-2.0  GNU General Public License, version 2 (GPL-2.0)
 */

chdir('../../');
require_once('includes/application_top.php');
$query = xtc_db_query("SELECT directory FROM " . TABLE_LANGUAGES . " WHERE code = '" . DEFAULT_LANGUAGE . "'");
$result = xtc_db_fetch_array($query);
require_once(DIR_WS_LANGUAGES . $result['directory'] . '/modules/payment/bitcoin.php');

$uniqid = filter_var($_GET['uniqid'], FILTER_SANITIZE_STRING);
$amount = filter_var($_GET['value'], FILTER_SANITIZE_NUMBER_INT);
$confirms = filter_var($_GET['confirmations'], FILTER_SANITIZE_NUMBER_INT);

if ($confirms < MODULE_PAYMENT_BITCOIN_API_CONFIRMS) {
    echo "*failed*";
} else {
    $query = xtc_db_query("SELECT orders_id FROM " . TABLE_ORDERS . "
                           WHERE bitcoin_uniqid = '" . $uniqid . "'
                             AND bitcoin_amount = '" . $amount . "'
                           LIMIT 1");
    $order = xtc_db_fetch_array($query);

    if(!$order) {
        echo "*failed*";
    } else {
        xtc_db_query("UPDATE " . TABLE_ORDERS . "
                      SET orders_status = '" . MODULE_PAYMENT_BITCOIN_PAID_STATUS . "'
                      WHERE orders_id = '" . $order['orders_id'] . "'");

        xtc_db_query("INSERT INTO " . TABLE_ORDERS_STATUS_HISTORY . "
                      (orders_id, orders_status_id, date_added, customer_notified, comments)
                      VALUES
                      ('" . $order['orders_id'] . "', '" . MODULE_PAYMENT_BITCOIN_PAID_STATUS . "',
                      now(), 1, '" . MODULE_PAYMENT_BITCOIN_PAID_COMMENT . "')");

        echo "*ok*";
    }
}

src/new/callback/bitcoin/exchange.php
<?php
/**
 * CommerceCoding Bitcoin Payment for Gambio GX2
 *
 * NOTICE OF LICENSE
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; version 2 of the License
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software
 * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
 *
 * @copyright   Copyright (c) 2013 CommerceCoding (http://www.commerce-coding.de)
 * @author      Alexander Diebler
 * @license     http://opensource.org/licenses/GPL-2.0  GNU General Public License, version 2 (GPL-2.0)
 */

chdir('../../');
require_once('includes/application_top.php');

switch (MODULE_PAYMENT_BITCOIN_SOURCE) {
    case 'blockchain.info':
        $json = file_get_contents('http://blockchain.info/ticker');
        $object = json_decode($json);
        $rate = $object->USD->sell;
        break;
    case 'bitstamp.net':
        $json = file_get_contents('https://www.bitstamp.net/api/ticker/');
        $object = json_decode($json);
        $rate = $object->ask;
        break;
    case 'coinbase.com':
        $json = file_get_contents('https://coinbase.com/api/v1/currencies/exchange_rates');
        $object = json_decode($json);
        $rate = $object->btc_to_usd;
        break;
    default:
        $rate = null;
}

if($rate) {
    $json = file_get_contents('https://bitdango.com/api/currencypairs/USDEUR');
    $object = json_decode($json);
    $usdToEur = $object->ExchangeRate;
    xtc_db_query("UPDATE " . TABLE_CONFIGURATION . " SET `configuration_value` = '" . number_format($rate * $usdToEur, 5, '.', '')  . "' WHERE `configuration_key` = 'MODULE_PAYMENT_BITCOIN_BTCEUR'");
}


src/new/lang/german/modules/payment/bitcoin.php
<?php
/**
 * CommerceCoding Bitcoin Payment for Gambio GX2
 *
 * NOTICE OF LICENSE
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; version 2 of the License
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software
 * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
 *
 * @copyright   Copyright (c) 2013 CommerceCoding (http://www.commerce-coding.de)
 * @author      Alexander Diebler
 * @license     http://opensource.org/licenses/GPL-2.0  GNU General Public License, version 2 (GPL-2.0)
 */

// Backend Information
define('MODULE_PAYMENT_BITCOIN_TEXT_TITLE', 'Bitcoin');
define('MODULE_PAYMENT_BITCOIN_TEXT_DESCRIPTION', 'Erlauben Sie ihren Kunden Bestellungen mit Bitcoins zu begleichen.');

// Configuration Titles & Descriptions
define('MODULE_PAYMENT_BITCOIN_STATUS_TITLE', 'Bitcoin Modul aktivieren');
define('MODULE_PAYMENT_BITCOIN_STATUS_DESC', 'M&ouml;chten Sie Zahlungen &uuml;ber Bitcoin akzeptieren?');
define('MODULE_PAYMENT_BITCOIN_ALLOWED_TITLE', 'Erlaubte Zonen');
define('MODULE_PAYMENT_BITCOIN_ALLOWED_DESC', 'Geben Sie <b>einzeln</b> die Zonen an, welche f&uuml;r dieses Modul erlaubt sein sollen. (z.B. AT,DE (wenn leer, werden alle Zonen erlaubt))');
define('MODULE_PAYMENT_BITCOIN_UNITS_TITLE', 'Einheiten');
define('MODULE_PAYMENT_BITCOIN_UNITS_DESC', 'In welchen Einheiten sollen Bitcoin-Preise angezeigt werden?');
define('MODULE_PAYMENT_BITCOIN_SOURCE_TITLE', 'Quelle f&uuml;r Bitcoin-Wechselkurse');
define('MODULE_PAYMENT_BITCOIN_SOURCE_DESC', '&Uuml;ber welchen Anbieter sollen Kursinformationen abgefragt werden?');
define('MODULE_PAYMENT_BITCOIN_BTCEUR_TITLE', 'Euro / Bitcoin');
define('MODULE_PAYMENT_BITCOIN_BTCEUR_DESC', 'Zu welchem Kurs sollen Euro in Bitcoin umgerechnet werden?');
define('MODULE_PAYMENT_BITCOIN_API_ADDRESS_TITLE', 'Ziel-Adresse');
define('MODULE_PAYMENT_BITCOIN_API_ADDRESS_DESC', 'Alle Zahlungen werden an diese Adresse weitergeleitet.');
define('MODULE_PAYMENT_BITCOIN_API_CONFIRMS_TITLE', 'Notwendige Best&auml;tigungen');
define('MODULE_PAYMENT_BITCOIN_API_CONFIRMS_DESC', 'Minimum der ben&ouml;tigten Best&auml;tigungen, um eine Zahlung zu akzeptieren.');
define('MODULE_PAYMENT_BITCOIN_API_SHARED_TITLE', 'Shared');
define('MODULE_PAYMENT_BITCOIN_API_SHARED_DESC', 'Die Zahlungen werden durch ein geteiltes Wallet geleitet, wodurch mehr Privatsphere erm&ouml;glicht werden soll. (Geb&uuml;hr: 0,5%) Mehr Informationen: http://blockchain.info/de/wallet/send-shared');
define('MODULE_PAYMENT_BITCOIN_NEW_STATUS_TITLE', 'Status f&uuml;r unbezahlte Bestellungen');
define('MODULE_PAYMENT_BITCOIN_NEW_STATUS_DESC', 'Geben Sie den Status an, welcher unbezahlten Bestellungen zugewiesen werden soll.');
define('MODULE_PAYMENT_BITCOIN_PAID_STATUS_TITLE', 'Status f&uuml;r bezahlte Bestellungen');
define('MODULE_PAYMENT_BITCOIN_PAID_STATUS_DESC', 'Geben Sie den Status an, welcher bezahlten Bestellungen zugewiesen werden soll.');
define('MODULE_PAYMENT_BITCOIN_SORT_ORDER_TITLE', 'Anzeigereihenfolge');
define('MODULE_PAYMENT_BITCOIN_SORT_ORDER_DESC', 'Reihenfolge der Anzeige. Kleinste Ziffer wird zuerst angezeigt.');

// Frontend Texts
define('MODULE_PAYMENT_BITCOIN_TEXT_FRONTEND_DESCRIPTION', 'Bezahlen Sie Ihre Bestellung mit Bitcoins. Der f&auml;llige Betrag wird in der Bestell&uuml;bersicht wieder angezeigt. Nach Abschluss Ihrer Bestellung erhalten Sie die Empf&auml;ngeradresse angezeigt. Die Waren werden nach Zahlungseingang versandt.');
define('MODULE_PAYMENT_BITCOIN_TEXT_ERROR', 'Zahlungsfehler');
define('MODULE_PAYMENT_BITCOIN_TEXT_PAYMENT_ERROR', 'Es gab einen Fehler bei der Erstellung einer Zahlungsadresse. Bitte versuchen Sie es erneut oder wechseln Sie die Zahlungsweise.');
define('MODULE_PAYMENT_BITCOIN_NEW_COMMENT', 'Bitcoin-Adresse: %s | Betrag: %s');
define('MODULE_PAYMENT_BITCOIN_PAID_COMMENT', 'Der Bitcoinbetrag wurde von der angegebenen Adresse empfangen und verifiziert.');